go/src/koding/kites/metrics/metrics.go

```diff
@@ -42,9 +42,14 @@ func NewWithPath(boltPath, app string) (*Metrics, error) {
 // NewWithDB creates a new metric collector with given bolt db instance. See New
 // func.
 func NewWithDB(db *bolt.DB, app string) (*Metrics, error) {
-	storage, err := NewBoltStorageWithDB(db)
+	var storage Storage
+	var err error
+	if db != nil {
+		storage, err = NewBoltStorageWithDB(db)
```

Comment:
This would make we would silently lose metrics.

If user `kd ssh some_machine`, the kd process is going to take a lock on metrics.bolt as long as the ssh session is alive.

If kd is executed concurrently (e.g. by atom or user in other terminal), it will fallback to in-memory store which would affectively drop any metrics created during its runtime.

Or I'm missing something?